flink-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/streaming/connectors/cassandra/CassandraSinkBase.java

```diff
@@ -85,7 +85,7 @@ public void onFailure(Throwable t) {
 			}
 		};
 		this.cluster = builder.getCluster();
-		this.session = cluster.connect();
+		this.session = cluster.connect(configuration.getString("keyspace", null));
```

Comment:
a cleaner approach would be to add a `protected Session createSession(Cluster cluster)` method. The default implementation returns `cluster.connect()`, and the pojo sink `cluster.connect(keyspace)`.

Reply:
indeed!